src/main/java/Duke.java

```diff
@@ -1,10 +1,291 @@
+        String taskNumber = line.substring(5);
+        int taskListElement = Integer.parseInt(taskNumber) - 1;
+        taskList.get(taskListElement).markAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println("[X] " + taskList.get(taskListElement).getDescription());
+        System.out.println(LINE);
+    }
+
+    private static void list() {
+        int taskCount = 1;
+        for (int i = 0; i < listCount; i++) {
+            System.out.println(taskCount + "." + taskList.get(i));
+            taskCount++;
+        }
+        System.out.println(LINE);
+    }
+
+    private static Todo todo(String line) {
+        char taskType = line.toUpperCase().charAt(0);
+        String taskDisplay = line.substring(5);
+
+        stringList.add(listCount, taskDisplay);
+        dueDate.add(listCount, null);
+        Todo todoTask = new Todo(taskDisplay, taskType);
+        taskList.add(listCount, todoTask);
+        return todoTask;
+    }
+
+    private static Deadline deadline(String line) {
+        int startingIndex = line.indexOf("/");
+        String taskDisplay = line.substring(9, startingIndex);
+        String doBy = "(" + line.substring(startingIndex + 1) + ")";
+        char taskType = line.toUpperCase().charAt(0);
+
+        stringList.add(listCount, taskDisplay);
+        dueDate.add(listCount, line.substring(startingIndex + 1));
+        Deadline deadlineTask = new Deadline(taskDisplay, taskType, doBy);
+        taskList.add(listCount, deadlineTask);
+        return deadlineTask;
+    }
+
+    private static Event event(String line) {
+        int startingIndex = line.indexOf("/");
+        String taskDisplay = line.substring(6, startingIndex);
+        String doBy = "(" + line.substring(startingIndex + 1) + ")";
+        char taskType = line.toUpperCase().charAt(0);
+
+        stringList.add(listCount, taskDisplay);
+        dueDate.add(listCount, line.substring(startingIndex + 1));
+        Event eventTask = new Event(taskDisplay, taskType, doBy);
+        taskList.add(listCount, eventTask);
+        return eventTask;
+    }
+
+    private static void printTask(Task task, int listCount) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(task);
+        System.out.println("Now you have " + listCount + " tasks in the list.");
+        System.out.println(LINE);
+    }
+
+    private static void delete(String line) {
+        int taskIndex = Integer.parseInt(line.substring(7)) - 1;
+        Task task = taskList.get(taskIndex);
+        listCount--;
+        System.out.println("Noted. I've removed this task:");
+        System.out.println(task);
+        System.out.println("Now you have " + listCount + " tasks in the list.");
+        System.out.println(LINE);
+        taskList.remove(taskIndex);
+        dueDate.remove(taskIndex);
+        stringList.remove(taskIndex);
+    }
+
+    private static void mainProgram(String line, Scanner in) {
```

Comment:
Consider shortening this method, as [it may be too long](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods).